src/v/storage/compacted_index_chunk_reader.cc

```diff
@@ -200,6 +202,9 @@ compacted_index_chunk_reader::load_slice(model::timeout_clock::time_point t) {
     if (!_cursor) {
         _cursor = ss::make_file_input_stream(_handle, 0, _footer->size);
     }
+    if (_as) {
+        _as->check();
+    }
```

Comment:
@mmaslankaprv i think this would be safer it wrapped the exception in an exceptional future. this isn't a coroutine, so that doesn't happen automatically. wdyt?